# **Dataset 3 — Power Consumption of Tetouan City (UCI)**

Fonte: https://archive.ics.uci.edu/dataset/849/power+consumption+of+tetouan+city

**Situação**

Uma equipe responsável pelo planejamento da distribuição de energia em Tétouan deseja identificar qual das três zonas apresenta o maior pico de consumo e observar as condições ambientais registradas nos momentos de maior demanda.

---
## Etapa A — Orange

Arquivo do fluxo: workflow_03.ows

**1. Carregue o dataset e visualize os registros.**
Widgets *CSV File Import* + *Data Table*: Tetuan City power consumption.csv — 52.416 registros, 9 atributos.

**2. Utilize Select Columns para manter as três variáveis de consumo, Temperature, Humidity e Wind Speed.**

**3. Observe no Data Table a escala de valores das três zonas de consumo.**
As três zonas estão na casa das **dezenas de milhares de kW**, enquanto os atributos ambientais ficam abaixo de 100.

**4. Verifique se existem valores ausentes.**
Widget *Feature Statistics*: coluna *Missing* = 0 em todos os atributos — não há valores ausentes.

**5. Gere uma amostra aleatória de 15% dos registros.**
Widget *Data Sampler*, proporção fixa de 15% — 7.863 registros.

**6. Exporte a amostra em CSV.**

---
## Etapa B — Python / Pandas

In [3]:
import pandas as pd

### Carregue a amostra no Pandas e apresente `head()`, `shape`, `info()` e `describe()`.

In [4]:
df = pd.read_csv('EX03_AMOSTRA_2.csv')
df.head()

,Zone 1 Power Consumption,Zone 2 Power Consumption,Zone 3 Power Consumption,Temperature,Humidity,Wind Speed
0,27335.69620,17872.34043,18442.40964,5.921,75.7,0.081
1,25998.98734,16993.31307,17945.06024,5.641,77.7,0.080
2,20621.77215,12853.49544,14105.06024,4.718,75.8,0.080
3,20858.73418,13203.64742,14538.79518,4.382,76.9,0.081
4,21776.20253,13626.74772,14625.54217,4.769,75.6,0.082


In [5]:
print(f'Total de registros: {df.shape[0]}')
print(f'Total de colunas: {df.shape[1]}')

Total de registros: 7863
Total de colunas: 6


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7863 entries, 0 to 7862
Data columns (total 6 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Zone 1 Power Consumption   7863 non-null   float64
 1   Zone 2  Power Consumption  7863 non-null   float64
 2   Zone 3  Power Consumption  7863 non-null   float64
 3   Temperature                7863 non-null   float64
 4   Humidity                   7863 non-null   float64
 5   Wind Speed                 7863 non-null   float64
dtypes: float64(6)
memory usage: 368.7 KB


In [7]:
df.describe()

,Zone 1 Power Consumption,Zone 2 Power Consumption,Zone 3 Power Consumption,Temperature,Humidity,Wind Speed
count,7863.000000,7863.000000,7863.000000,7863.000000,7863.000000,7863.000000
mean,32371.549651,21060.734034,17845.600096,18.789376,68.400612,1.965602
std,7134.686966,5202.357443,6573.203879,5.864564,15.665773,2.350954
min,14612.658230,9436.252546,6211.764706,3.541000,13.340000,0.054000
25%,26263.115455,17013.383055,13200.729480,14.340000,58.485000,0.078000
50%,32249.429660,20851.559250,16490.322580,18.690000,70.000000,0.086000
75%,37421.355555,24812.249210,21763.603550,22.820000,81.500000,4.915000
max,51571.498340,36437.170010,47441.673640,39.600000,94.700000,6.325000


---
### 1. Renomeie as três variáveis de consumo para Consumo_Zona_1, Consumo_Zona_2 e Consumo_Zona_3.

In [8]:
df.columns

Index(['Zone 1 Power Consumption', 'Zone 2  Power Consumption',
       'Zone 3  Power Consumption', 'Temperature', 'Humidity', 'Wind Speed'],
      dtype='object')

In [9]:
df.rename(columns={'Zone 1 Power Consumption':  'Consumo_Zona_1',
                   'Zone 2  Power Consumption': 'Consumo_Zona_2',
                   'Zone 3  Power Consumption': 'Consumo_Zona_3',
                   }, inplace=True)

df.head()

,Consumo_Zona_1,Consumo_Zona_2,Consumo_Zona_3,Temperature,Humidity,Wind Speed
0,27335.69620,17872.34043,18442.40964,5.921,75.7,0.081
1,25998.98734,16993.31307,17945.06024,5.641,77.7,0.080
2,20621.77215,12853.49544,14105.06024,4.718,75.8,0.080
3,20858.73418,13203.64742,14538.79518,4.382,76.9,0.081
4,21776.20253,13626.74772,14625.54217,4.769,75.6,0.082


---
### 2. Determine o consumo máximo registrado em cada uma das três zonas.

In [12]:
zonas = ['Consumo_Zona_1', 'Consumo_Zona_2', 'Consumo_Zona_3']

max_consumo = df[zonas].max()
print(max_consumo)

Consumo_Zona_1    51571.49834
Consumo_Zona_2    36437.17001
Consumo_Zona_3    47441.67364
dtype: float64


---
### 3. Identifique qual zona apresenta o maior pico de consumo na amostra.

In [13]:
zona_pico  = max_consumo.idxmax()
valor_pico = max_consumo.max()

print(f'A zona com o maior pico de consumo e: {zona_pico}')
print(f'Valor do pico: {valor_pico:.2f} kW')

A zona com o maior pico de consumo e: Consumo_Zona_1
Valor do pico: 51571.50 kW


---
### 4. Para a zona identificada, calcule 70% do valor máximo e crie um DataFrame com os registros acima desse limiar.

In [14]:
limiar  = valor_pico * 0.70
df_alto = df[df[zona_pico] > limiar]

print(f'Limiar (70% de {valor_pico:.2f}): {limiar:.2f} kW')
print()
df_alto.head()

Limiar (70% de 51571.50): 36100.05 kW



,Consumo_Zona_1,Consumo_Zona_2,Consumo_Zona_3,Temperature,Humidity,Wind Speed
12,39517.97468,24463.22188,25619.27711,14.29,63.94,0.078
13,39596.96203,24000.00000,25798.55422,14.09,64.87,0.078
14,39991.89873,24477.81155,25989.39759,12.51,68.35,0.077
15,39141.26582,24900.91185,25607.71084,11.49,73.10,0.077
33,40648.10127,26764.74164,24734.45783,14.81,61.79,0.078


---
### 5. Conte os registros e calcule o percentual que eles representam.

In [15]:
qtd = len(df_alto)
pct = qtd / len(df) * 100

print(f'Quantidade de registros acima do limiar: {qtd}')
print(f'Percentual da amostra: {pct:.2f}%')

Quantidade de registros acima do limiar: 2388
Percentual da amostra: 30.37%


---
### 6. Calcule a temperatura média e crie um segundo DataFrame contendo consumo acima do limiar e temperatura acima da média.

In [16]:
media_temp     = df['Temperature'].mean()
df_alto_quente = df_alto[df_alto['Temperature'] > media_temp]

qtd2 = len(df_alto_quente)
pct2 = qtd2 / len(df) * 100

print(f'Temperatura media da amostra: {media_temp:.2f} C')
print(f'Registros com consumo alto E temperatura acima da media: {qtd2}')
print(f'Percentual da amostra: {pct2:.2f}%')

Temperatura media da amostra: 18.79 C
Registros com consumo alto E temperatura acima da media: 1597
Percentual da amostra: 20.31%


In [17]:
df_alto_quente.head()

,Consumo_Zona_1,Consumo_Zona_2,Consumo_Zona_3,Temperature,Humidity,Wind Speed
1337,37421.61702,21662.19512,20409.67742,19.76,56.90,4.915
1338,43457.36170,25569.51220,25072.25806,19.27,56.40,4.916
1829,37550.29787,19481.70732,16385.80645,22.17,56.91,0.085
1837,36949.78723,19302.43902,22883.22581,23.94,48.92,0.082
1838,44731.91489,23315.85366,27673.54839,22.08,57.60,0.086


---
### 7. Compare os dois conjuntos e explique o que ocorreu com a quantidade de registros após a inclusão da condição ambiental.

In [18]:
reducao   = qtd - qtd2
retencao  = qtd2 / qtd * 100
base      = (df['Temperature'] > media_temp).mean() * 100

print(f'Conjunto 1 - consumo alto {qtd} registros ({pct:.2f}% da amostra)')
print(f'Conjunto 2 - consumo alto + calor {qtd2} registros ({pct2:.2f}% da amostra)')
print()
print(f'Registros removidos pela condicao ambiental: {reducao}')
print(f'Percentual de picos que ocorrem em alta temperatura: {retencao:.2f}%')
print(f'Percentual da amostra toda com temperatura acima da media: {base:.2f}%')

Conjunto 1 - consumo alto 2388 registros (30.37% da amostra)
Conjunto 2 - consumo alto + calor 1597 registros (20.31% da amostra)

Registros removidos pela condicao ambiental: 791
Percentual de picos que ocorrem em alta temperatura: 66.88%
Percentual da amostra toda com temperatura acima da media: 49.51%


**Resposta:**

**O que ocorreu:** a quantidade caiu de **2.388 para 1.597** registros 791 a menos. A queda era esperada, porque o segundo critério é aplicado com o conector e dentro do primeiro conjunto: o conjunto 2 é necessariamente um subconjunto do conjunto 1 e só pode encolher.

**O que o resultado mostra:** o número relevante não é a queda, e sim a proporção que sobrou. Na amostra inteira, **49,51%** dos registros têm temperatura acima da média praticamente metade, o que é esperado por definição de média. Já entre os registros de alto consumo, esse percentual sobe para **66,88%**.

Ou seja, os picos de demanda da Zona 1 se concentram nos momentos mais quentes, o que aponta a carga térmica (refrigeração) como componente relevante da ponta. Mas o calor não explica todos eles: **33,12%** dos picos ocorrem com temperatura abaixo da média. A temperatura é um fator associado à alta demanda, não a sua única causa.